bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableAdmin.java

```diff
@@ -866,46 +865,31 @@ public boolean isTableAvailable(TableName tableName, byte[][] splitKeys) throws
   public void snapshot(String snapshotName, TableName tableName)
       throws IOException, SnapshotCreationException, IllegalArgumentException {
 
-    Operation operation = snapshotTable(snapshotName, tableName);
-    try {
-      connection.getSession().getInstanceAdminClient().waitForOperation(operation);
-    } catch (TimeoutException e) {
-      throw new IOException("Timed out waiting for snapshot creation to finish", e);
-    }
+    snapshotTable(snapshotName, tableName);
   }
 
-  /**
-   * Creates a snapshot from an existing table. NOTE: Cloud Bigtable has a cleanup policy
-   *
-   * @param snapshotName a {@link String} object.
-   * @param tableName a {@link TableName} object.
-   * @return a {@link Operation} object.
-   * @throws IOException if any.
-   */
-  protected Operation snapshotTable(String snapshotName, TableName tableName) throws IOException {
-    Preconditions.checkNotNull(snapshotName);
-    Preconditions.checkNotNull(tableName);
-    if (snapshotName.isEmpty()) {
-      // HBase returns an empty operation instance in case snapshotName is an empty string.
-      return Operation.newBuilder().build();
+  protected Backup snapshotTable(String snapshotName, TableName tableName) throws IOException {
+    if (Strings.isNullOrEmpty(snapshotName)) {
```

Comment:
I think Preconditions.checkArgument communicates your intention better than if statements

Reply:
Also, can you try to use **Id** suffix for short "backupId" and **Name** for long "projects/projectId/instances/instanceId/clusters/clusterId/backups/backupId"